relay-profiling/src/sample.rs

```diff
@@ -213,6 +213,13 @@ impl SampleProfile {
         true
     }
 
+    fn check_duration(&self) -> bool {
+        if let Some(sample) = &self.profile.samples.last() {
+            return sample.elapsed_since_start_ns <= 30_000_000_000;
```

Comment:
nit: Would be nice if we could use the const declared in the android module and do something like `THIRTY_SECONDS.as_nanos()`